To: vim_dev@googlegroups.com
Subject: Patch 7.3.126
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.126
Problem:    Compiler warning for signed pointer.
Solution:   Use unsigned int argument for sscanf().
Files:	    src/blowfish.c


*** ../vim-7.3.125/src/blowfish.c	2010-08-15 21:57:28.000000000 +0200
--- src/blowfish.c	2010-12-17 19:58:18.000000000 +0100
***************
*** 413,418 ****
--- 413,419 ----
      int    salt_len;
  {
      int      i, j, keypos = 0;
+     unsigned u;
      UINT32_T val, data_l, data_r;
      char_u   *key;
      int      keylen;
***************
*** 432,439 ****
      }
      for (i = 0; i < keylen; i++)
      {
! 	sscanf((char *)&key[i * 2], "%2x", &j);
! 	key[i] = j;
      }
  
      mch_memmove(sbx, sbi, 4 * 4 * 256);
--- 433,440 ----
      }
      for (i = 0; i < keylen; i++)
      {
! 	sscanf((char *)&key[i * 2], "%2x", &u);
! 	key[i] = u;
      }
  
      mch_memmove(sbx, sbi, 4 * 4 * 256);
*** ../vim-7.3.125/src/version.c	2011-02-15 18:06:11.000000000 +0100
--- src/version.c	2011-02-25 14:40:22.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     126,
  /**/

-- 
Why doesn't Tarzan have a beard?

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
